build.sh

```diff
@@ -1,73 +1,77 @@
 #!/bin/bash
-#
-#
 
-RELEASE_VERSION=
-buildpath=
+set -e
 
-function setuptree() {
-  b=$( mktemp -d $buildpath/gh-ostXXXXXX ) || return 1
-  mkdir -p $b/gh-ost
-  mkdir -p $b/gh-ost/usr/bin
-  echo $b
-}
-
-function build {
-  osname=$1
-  osshort=$2
-  GOOS=$3
-  GOARCH=$4
+. ./script/common
+buildpath=build
+timestamp=$(date "+%Y%m%d%H%M%S")
 
+function build() {
+  goos=$1
+  osshort=$goos
 
+  envs="-e GOOS=$goos \
+    -e GOARCH=amd64 \
+    -e CGO_ENABLED=1 \
+    -e GOCACHE=/go/src/${PACKAGE_PATH}/.cache \
+    -e RELEASE_VERSION=${RELEASE_VERSION} \
+    -e BIN_PATH=${buildpath}/bin"
 
-  if ! go version | egrep -q 'go(1[.]9|1[.]1[0-9])' ; then
-    echo "go version is too low. Must use 1.9 or above"
-    exit 1
+  if [ "$goos" == "darwin" ]; then
+    # specify cross compiler
+    envs="$envs -e CC=x86_64h-apple-darwin14-cc"
```

Comment:
This is the magic of `dockercore/golang-cross`. By specifying the `CC` that's already bundled, we can cross compile for darwin.

I'm 90% sure this is fine for the release. I copied this over to a mac machine I have and verified that I can run at least `--version`. (I might eat my hat once the localtests, but I'm pretty confident this will be okay)